### PR TITLE
Task/white238/update uberenv

### LINKED ISSUE #350: Double call to uberenv causes incoherency in the logs of Gitlab CI.
See https://lc.llnl.gov/gitlab/smith/serac/-/jobs/161492/raw

### PR DESCRIPTION
**Pull request by white238**
* Move devtools configs to their own directories
* Update system packages for devtools packages.yaml
* Bump spack version for devtools builds
* Sync build scripts w/ changes in axom for less differences between two repos
* Fix bug where compiler spec wasnt passed to mirror commands

### CODE REVIEW

**Comment by white238** [comment]
LGTM

File: scripts/llnl/common_build_functions.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def assertUberenvExists():
         sys.exit(1)
 
 
-def uberenv_create_mirror(prefix, project_file, mirror_path):
+def uberenv_create_mirror(prefix, spec, project_file, mirror_path):
```
Comment (joshessman-llnl) [inline scripts/llnl/common_build_functions.py:136]: I might be missing something but is the spec used anywhere in this function?

**Comment by white238** [comment]
LGTM

File: scripts/spack/devtools.json [hunk 2]
```diff
@@ -1,9 +1,11 @@
 {
 "package_name" : "serac_devtools",
 "package_version" : "fakeversion",
-"package_final_phase" : "",
 "package_source_dir" : "../..",
+"package_final_phase" : "",
 "spack_url": "https://github.com/spack/spack",
-"spack_commit": "ccf94ded67e688770dc8d0388d8f75d0d6641e1e",
+"spack_commit": "e0e6a0199ef4db473ea0bd2e924f3e96fa10297a",
+"spack_configs_path": "scripts/spack/devtools_configs",
```
Comment (jamiebramwell) [inline scripts/spack/devtools.json:8]: 👍

File: scripts/spack/devtools_configs/toss_3_x86_64_ib/packages.yaml [hunk 3]
```diff
@@ -89,15 +115,15 @@ packages:
     externals:
     - spec: pkg-config
       prefix: /usr
-  tar:
+  readline:
```
Comment (jamiebramwell) [inline scripts/spack/devtools_configs/toss_3_x86_64_ib/packages.yaml:118]: This looks like there's a story behind it...

**Review by jamiebramwell — APPROVED** [review]

File: scripts/spack/devtools_configs/toss_3_x86_64_ib/packages.yaml (hunk 3 above)
Comment (white238) [reply scripts/spack/devtools_configs/toss_3_x86_64_ib/packages.yaml:118]: spack built like 3x the amount of packages as before...

Comment (joshessman-llnl) [reply scripts/spack/devtools_configs/toss_3_x86_64_ib/packages.yaml:118]: Just wait till you see all the changes that were needed to the `packages.yaml` to work with Spack's new concretizer...

File: scripts/llnl/common_build_functions.py [hunk 4]
```diff
@@ -133,24 +133,29 @@ def assertUberenvExists():
...
     """
     Calls uberenv to create a spack mirror.
     """
     assertUberenvExists()
     cmd  = "python {0} --create-mirror -k ".format(get_uberenv_path())
     cmd += "--prefix=\"{0}\" --mirror=\"{1}\" ".format(prefix, mirror_path)
+    cmd += "--spec=\"{0}\" ".format(spec)
```
Comment (joshessman-llnl) [inline scripts/llnl/common_build_functions.py:143]:
```suggestion
    cmd += " --spec=\"{0}\" ".format(spec)
```

File: scripts/llnl/common_build_functions.py [hunk 5]
```diff
@@ -133,24 +133,29 @@ def assertUberenvExists():
...
     if project_file:
         cmd += "--project-json=\"{0}\" ".format(project_file)
-    res = sexe(cmd, echo=True, error_prefix="WARNING:")
+
     print("[~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~]")
     print("[ It is expected for 'spack --create-mirror' to throw warnings.                ]")
     print("[~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~]")
+    res = sexe(cmd, echo=True, error_prefix="WARNING:")
+    print("[~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~]")
+    print("[ End of expected warnings from 'spack --create-mirror'                        ]")
```
Comment (joshessman-llnl) [inline scripts/llnl/common_build_functions.py:152]: 👍 this is useful, it's not always easy to tell where this stage ends

**Review by joshessman-llnl — APPROVED** [review]

**Pull request merged**